SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -463,12 +482,58 @@ void UGlobalStateManager::SetCanBeginPlay(bool bInCanBeginPlay)
 	NetDriver->Connection->SendComponentUpdate(GlobalStateManagerEntityId, &Update);
 }
 
+void UGlobalStateManager::SetAuthBeginPlayCalled(const bool bInAuthBeginPlayCalled)
+{
+	if (!NetDriver->StaticComponentView->HasAuthority(GlobalStateManagerEntityId, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID))
+	{
+		UE_LOG(LogGlobalStateManager, Warning, TEXT("Tried to set AuthBeginPlayCalled on the GSM but this worker does not have authority."));
+		return;
+	}
+
+	Worker_ComponentUpdate Update = {};
+	Update.component_id = SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID;
+	Update.schema_type = Schema_CreateComponentUpdate(SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID);
+	Schema_Object* UpdateObject = Schema_GetComponentUpdateFields(Update.schema_type);
+
+	Schema_AddBool(UpdateObject, SpatialConstants::STARTUP_ACTOR_MANAGER_AUTH_BEGIN_PLAY_CALLED_ID, static_cast<uint8_t>(bInAuthBeginPlayCalled));
+
+	bAuthBeginPlayCalled = bInAuthBeginPlayCalled;
+	NetDriver->Connection->SendComponentUpdate(GlobalStateManagerEntityId, &Update);
+}
+
 void UGlobalStateManager::TryTriggerBeginPlay()
 {
+	// clients should not trigger begin play via this pathway
+	check(NetDriver->IsServer());
+
 	if (bCanBeginPlay && NetDriver->EntityPool != nullptr && NetDriver->EntityPool->IsReady())
 	{
+		const bool bStartupActorAuthServer = NetDriver->StaticComponentView->HasAuthority(GlobalStateManagerEntityId, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID);
+		const bool bCallBeginPlayWithAuthority = bStartupActorAuthServer && !bAuthBeginPlayCalled;
+
+		TArray<ActorAuthCapture> OutOriginalActorRoles;
+		const ENetRole RoleToSet = bCallBeginPlayWithAuthority ? ROLE_Authority : ROLE_SimulatedProxy;
+		const ENetRole RemoteRoleToSet = bCallBeginPlayWithAuthority ? ROLE_SimulatedProxy : ROLE_Authority;
+
+		// CMS_TODO: Is this approach ok, or should we only make sure Role is set as necessary,
+		// and preserve the value of RemoteRole, rather than setting it to SimulatedProxy?  Bigger issue when calling
+		// with authority, since we don't restore original roles in that case.
+		SetRolesOnAllReplicatedActors(RoleToSet, RemoteRoleToSet, &OutOriginalActorRoles);
+
 		TriggerBeginPlay();
```

Comment:
Note to me: It's possible that calling BeginPlay() on an actor results in a new actor being created with authority setting that don't match what we've setup, and it can gets it's own call to BeginPlay() immediately during the iteration over world actors in NotifyBeginPlay().

Reply:
this case is handled with the latest upload